dbt_semantic_interfaces/protocols/metric.py

```diff
@@ -212,6 +212,11 @@ def metrics(self) -> Optional[Sequence[MetricInput]]:  # noqa: D
     def conversion_type_params(self) -> Optional[ConversionTypeParams]:  # noqa: D
         pass
 
+    @property
+    @abstractmethod
+    def conversion_type_params_or_error(self) -> ConversionTypeParams:  # noqa: D
```

Comment:
Oof. I don't think we want this on the type params protocol.

If this is a real problem we can add a helper like `assert_and_return(input: Optional[T]) -> T` inside MetricFlow, or if we're doing this in a handful of places adding an `assert metric.type_params.conversion_type_params is not None` call before accessing the property. I'm pretty sure MyPy is currently smart enough to refine the return type on that property for us.

Reply:
➕ ➕ We've had this logic crop up before, and I'd prefer us to push it out of DSI. It seems unnecessary to have on the protocols, and the DSI provided implementations.

Reply:
yea it was just annoying having to assert, but makes sense that we sohuld keep it out of DSI. I'll probably do something similar to what Tom suggested in MF